### PR TITLE
 tools: ceph-authtool: report correct number of caps when creating keyring

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: https://tracker.ceph.com/issues/23776

### CODE REVIEW

**Comment by smithfarm** [comment]
Test the reproducer:

`teuthology-suite -k distro --machine-type mira --email ncutler@suse.cz --priority 90 --ceph-repo https://github.com/ceph/ceph.git --ceph master --newest 10 --suite-repo https://github.com/smithfarm/ceph.git --suite-branch wip-23776 --suite rados/singleton-nomsgr --filter=ceph_authtool_caps`

**running** http://pulpito.ceph.com/smithfarm-2018-07-29_17:52:43-rados:singleton-nomsgr-master-distro-basic-mira/

**Comment by smithfarm** [comment]
@ErwanAliasr1 ccache kicked in here - the last "make check" build went incredibly fast. I couldn't believe my eyes!

```
cache directory                     /home/jenkins-build/.ccache
primary config                      /home/jenkins-build/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
cache hit (direct)                  1438
cache hit (preprocessed)               0
cache miss                             6
called for link                      335
unsupported source language           82
files in cache                      2948
cache size                           8.7 GB
max cache size                     100.0 GB
```

**Comment by ErwanAliasr1** [comment]
Glad it saves time !

Le dim. 29 juil. 2018 à 22:09, Nathan Cutler <notifications@github.com> a
écrit :

**Comment by smithfarm** [comment]
test this please

File: src/test/cli/ceph-authtool/add-key.t [hunk 1]
```diff
@@ -2,7 +2,7 @@
   creating kring
 
   $ ceph-authtool kring --add-key 'AQAK7yxNeF+nHBAA0SgSdbs8IkJrxroDeJ6SwQ== 18446744073709551615'
-  added entity client.admin auth auth(auid = 18446744073709551615 key=AQAK7yxNeF+nHBAA0SgSdbs8IkJrxroDeJ6SwQ== with 0 caps)
+  added entity client.admin auth(auid = 18446744073709551615 key = AQAK7yxNeF+nHBAA0SgSdbs8IkJrxroDeJ6SwQ==)
```
Comment (smithfarm) [inline src/test/cli/ceph-authtool/add-key.t]: maybe add a comma to better separate `auid` from `key`? i.e.

`added entity client.admin auth(auid = 18446744073709551615, key = AQAK7yxNeF+nHBAA0SgSdbs8IkJrxroDeJ6SwQ==)`

?

**Comment by smithfarm** [comment]
jenkins test docs

**Comment by smithfarm** [comment]
@gregsfortytwo Had to make some larger changes in the rebase (auid is gone now) - can you take another look?

**Pull request merged**